packaging/flavorgen/cloudprovider/cloud-controller-manager.go

```diff
@@ -35,295 +35,63 @@ const (
 	CPIControllerImageURL = "gcr.io/cloud-provider-vsphere/cpi/release/manager"
 )
 
-// CloudControllerManagerServiceAccount returns the ServiceAccount used for the cloud-controller-manager.
-func CloudControllerManagerServiceAccount() *corev1.ServiceAccount {
-	return &corev1.ServiceAccount{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "cloud-controller-manager",
-			Namespace: "kube-system",
-		},
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "ServiceAccount",
-			APIVersion: corev1.SchemeGroupVersion.String(),
-		},
+// CloudControllerManagerManifests returns the yaml manifests generated via
+// `helm template` from https://github.com/kubernetes/cloud-provider-vsphere/tree/master/charts/vsphere-cpi
+func CloudControllerManagerManifests() (string, error) {
+	objectList := []string{}
+
+	imageTagRemoval := regexp.MustCompile(`(image: .*):.*`)
+	withoutTag := imageTagRemoval.ReplaceAll([]byte(cpiManifests), []byte(`${1}:$`+env.CPIImageKubernetesVersionVar))
+
+	// ClusterResourceSet does not support `kind: List`, we have to adjust the manifests
+	// to remove those lists and directly add the items instead.
```

Comment:
Do you remember what error you were getting?

I wonder if it is as easy as adding List to the schema that CRS uses

Reply:
```
E0122 17:09:29.089263   18954 clusterresourceset_controller.go:362] "failed to apply ClusterResourceSet resource" err="creating object /v1, Kind=List : no matches for kind \"\" in version \"v1\"" controller="clusterresourceset" controllerGroup="addons.cluster.x-k8s.io" controllerKind="ClusterResourceSet" ClusterResourceSet="test-cluster-resource-set-7xskb/clusterresourceset-hybis4" namespace="test-cluster-resource-set-7xskb" name="clusterresourceset-hybis4" reconcileID="030779b8-e73c-44e3-bde7-9948b82e6455" Cluster="test-cluster-resource-set-7xskb/cluster-sdk18p" Resource kind="ConfigMap" Resource name="test-configmap"
```

Reply:
> Keep in mind that the Kubernetes API does not have a kind named List.
> 
> kind: List is a client-side, internal implementation detail for processing collections that might be of different kinds of object. Avoid depending on kind: List in automation or other code.

Source: https://kubernetes.io/docs/reference/using-api/api-concepts/

Reply:
Ah perfect. Okay that tells us we shouldn't fix it :D